spec/vector_spec.rb

```diff
@@ -1769,6 +1769,24 @@
     end
   end
 
+  context '#rolling_fillna!' do
+    subject do
+      Daru::Vector.new(
+        [1, 2, 1, 4, nil, Float::NAN, 3, nil, Float::NAN]
+      )
+    end
+
+    context 'rolling_fillna! forwards' do
+      before { subject.rolling_fillna!(:forward) }
+      its(:to_a) { is_expected.to eq [1, 2, 1, 4, 4, 4, 3, 3, 3] }
+    end
+
+    context 'rolling_fillna! backwards' do
+      before { subject.rolling_fillna!(direction: :backward) }
+      its(:to_a) { is_expected.to eq [1, 2, 1, 4, 3, 3, 3, 0, 0] }
+    end
```

Comment:
Don't forget about edge cases (vector starting with non-value, vector entirely consisting of non-values and so on)